formula-test/src/main/java/com/instacart/formula/test/TestFormula.kt

```diff
@@ -3,31 +3,18 @@ package com.instacart.formula.test
 import com.instacart.formula.Action
 import com.instacart.formula.Evaluation
 import com.instacart.formula.Formula
+import com.instacart.formula.IFormula
 import com.instacart.formula.Snapshot
 import java.util.concurrent.atomic.AtomicLong
 
 /**
  * Test formula is used to provide a fake formula implementation. It allows you to [send][output]
  * output updates and [inspect/interact][input] with input.
  */
-abstract class TestFormula<Input, Output> :
```

Comment:
Having both inheritance and factory approaches creates inconsistencies in the API. I've decided to make this a final class and keep the following API as the primary approach 
```kotlin
class FakeUserFormula : UserFormula {
     override val implementation = testFormula(
         initialOutput = User(
             name = "Fake user name",
         )
     )
 }
```